cxx-idioms/named_params_2.cpp

```diff
@@ -0,0 +1,57 @@
+#include <iostream>
+#include <string>
+#include <tuple>
+
+template <class T>
+struct make_param
+{
+  make_param(T&& o): v(std::move(o)) { }
+  make_param(T const& o): v(o) { }
+  operator T() { return std::move(v); }
```

Comment:
this move looks dangerous to me. I guess the idea is that this operator is only used once?